### PR TITLE
feature: update command for api project

### LINKED ISSUE #85: Create an "update" command for API that will update the package.json/plugins.json to the latest versions
This should check against the current package.json in the released version and interactively prompt the user if they want to update each plugin.

### PR DESCRIPTION
**Pull request by vanpho93**
Resolves: https://github.com/reactioncommerce/cli/issues/85

# Solution
Read local package.json and reaction package.sjon on trunk branch,
Compare and update packages
Run npm install

# Testing
inside cli run `npm install -g`
then inside you cli project run `reaction update`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -479,7 +479,8 @@
     "error",
     "double",
     {
-      "avoidEscape": true
+      "avoidEscape": true,
+      "allowTemplateLiterals": true
```
Comment (brent-hoover) [inline .eslintrc]: Why are we making this change? Generally we should never touch this

File: tests/utils/execute.js [hunk 2]
```diff
@@ -9,13 +9,15 @@ import createProcess from "./createProcess.js";
...
-  const { env = null } = opts;
-  const childProcess = createProcess(processPath, args, env);
+  const { env = null, cwd = null, agnoreError = false } = opts;
+  const childProcess = createProcess(processPath, args, env, cwd);
   childProcess.stdin.setEncoding("utf-8");
   const promise = new Promise((resolve, reject) => {
-    childProcess.stderr.once("data", (err) => {
-      reject(err.toString());
-    });
+    if (agnoreError === false) {
```
Comment (brent-hoover) [inline tests/utils/execute.js]: Is this maybe supposed to be `ignoreError` rather than `agnoreError`

**Review by brent-hoover — COMMENTED** [review]
Looks great! Just a couple of nitpicky questions

File: .eslintrc (hunk 1 above)
Comment (vanpho93) [reply .eslintrc]: I used allowTemplateLiterals in here `const invalidPackageJson = `{ "name": "test }`;`
And eslint not allow that.

File: tests/utils/execute.js (hunk 2 above)
Comment (vanpho93) [reply tests/utils/execute.js]: I fixed.

File: .eslintrc (hunk 1 above)
Comment (vanpho93) [reply .eslintrc]: Do you think we should revert this change?

Comment (brent-hoover) [reply .eslintrc]: Yes, when you encounter a discrepancy between linting and your code the path is to correct your code, not the lint rules. The lint rules are the same for the entire organization.

Comment (vanpho93) [reply .eslintrc]: Understood. I reverted.

**Review by brent-hoover — APPROVED** [review]

**Comment by brent-hoover** [comment]
@tedraykov Second call for PR review

File: commands/update.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
+import fs from "fs/promises";
+import { execSync } from "child_process";
+import inquirer from "inquirer";
+import compareVerions from "compare-versions";
```
Comment (tedraykov) [inline commands/update.js]: It's a typo
```suggestion
import compareVersions from "compare-versions";
```

File: commands/update.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+  Logger.info("Getting the remote dependencies");
+  const remoteDependencies = await getRemoteDependencies();
+
+  const outdatedPackages = pluginPackages
+    .map(({ name, version }) => ({
+      name,
+      version,
+      remoteVersion: remoteDependencies[name],
+      outdated: compareVerions.compare(cleanVersion(remoteDependencies[name]), cleanVersion(version), ">")
```
Comment (tedraykov) [inline commands/update.js]: Rename the default import name here as well.

File: commands/update.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  Logger.info(`Found ${pluginPackages.length} plugins. ${outdatedPackages.length} outdated plugins.`);
+
+
+  const answers = options.all ? null : await inquirer.prompt([
+    {
+      type: "list",
+      name: "kind",
+      message: "Do want to update your outdate plugins?",
```
Comment (tedraykov) [inline commands/update.js]: Outdate is a verb, so `outdated` is the correct spelling here.
```suggestion
      message: "Do want to update your outdated plugins?",
```

**Review by tedraykov — CHANGES_REQUESTED** [review]
Overall great job. I added a few comments about typos only. Apart from them, it's good to go.

**Review by tedraykov — APPROVED** [review]

**Pull request merged**